src/Plugin/Field/FieldFormatter/AdvancedEntityReferenceFormatter.php

```diff
@@ -38,48 +40,93 @@ public static function defaultSettings() {
   public function settingsForm(array $form, FormStateInterface $form_state) {
     $elements = parent::settingsForm($form, $form_state);
     $elements['view_mode']['#title'] = $this->t('Default view mode');
+    $target_type = $this->getFieldSetting('target_type');
+    $handler_settings = $this->getFieldSetting('handler_settings');
+    $target_bundles = $handler_settings['target_bundles'];
+    $view_modes = $this->entityDisplayRepository->getViewModeOptions($target_type);
+    $field_name = $this->fieldDefinition->getName();
+    $settings = $this->getSettings();
+    $elements['override'] = [
+      '#type' => 'radios',
+      '#options' => [
+        'none' => $this->t('None'),
+        'bundle' => $this->t('Select view modes based on target bundle'),
+        'delta' => $this->t('Select view modes based on item delta'),
+      ],
+      '#title' => $this->t('Override options'),
+      '#default_value' => $settings['override'],
+    ];
+    $elements['bundle_view_mode'] = [
+      '#type' => 'container',
+      '#attributes' => ['id' => 'bundle-view-mode-wrapper'],
+      '#states' => [
+        'visible' => [
+          ':input[name="fields[' . $field_name . '][settings_edit_form][settings][override]"]' => ['value' => 'bundle'],
+        ],
+      ],
+    ];
+    foreach ($target_bundles as $bundle) {
+      $elements['bundle_view_mode'][$bundle] = [
+        '#type' => 'select',
+        '#options' => $view_modes,
```

Comment:
Need to use `getViewModeOptionsByBundle()` method to show only view modes for specific bundle